core/include/thread_flags.h

```diff
@@ -97,9 +97,12 @@
  *      }
  */
 #define THREAD_FLAG_MSG_WAITING      (0x1<<15)
-
-#define THREAD_FLAG_MUTEX_UNLOCKED   (0x1<<14)
-#define THREAD_FLAG_TIMEOUT          (0x1<<13)
+/**
+ * @brief Thread flag which is set by xtimer_set_timeout_flag when the timer expires
```

Comment:
(minor, feel free to change)

The other flag has "Set when ..." as brief. IMO "Set by @ref xtimer_set_timeout_flag() when the timer expires" would fit better here, for more consistent look in the generated doxygen.